### PR TITLE
[MINOR][DOC] Add more details to descriptions of customized class path

### PR DESCRIPTION
**Pull request by lys0716**
## What changes were proposed in this pull request?

Add more details to descriptions of customized class path

## How was this patch tested?

N/A

Signed-off-by: Yishuang Lu <luystu@gmail.com>

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -441,7 +441,7 @@ Apart from these, the following properties are also available, and may be useful
   <td>false</td>
   <td>
     (Experimental) Same functionality as <code>spark.driver.userClassPathFirst</code>, but
-    applied to executor instances.
+    applied to executor instances. This is used in both cluster mode and client mode.
```
Comment (srowen) [inline docs/configuration.md]: This seems superfluous; executor classpath handling shouldn't vary from client to cluster mode here anyway?

File: docs/configuration.md [hunk 2]
```diff
@@ -365,7 +365,7 @@ Apart from these, the following properties are also available, and may be useful
   <td>
     Extra classpath entries to prepend to the classpath of executors. This exists primarily for
     backwards-compatibility with older versions of Spark. Users typically should not need to set
-    this option.
+    this option. Also see <code>spark.executor.userClassPathFirst</code> to prepend jar ahead of classpath of spark.
```
Comment (srowen) [inline docs/configuration.md]: I don't think this clarifies; it doesn't cause these classpath entries (not jars) to be prepended, but to take precedence over Spark's. That's already stated in the docs though.

File: docs/configuration.md (hunk 1 above)
Comment (lys0716) [reply docs/configuration.md]: The reason is that the descriptions of <code>spark.driver.userClassPathFirst</code> said **This is used in cluster mode only**. But <code>spark.executor.userClassPathFirst</code> works for client mode as well.

In our case, we use client mode. There is a use case that we want to overwrite executor class path. We try <code>spark.executor.extraClassPath</code>, but it is not enough. I had to use <code>spark.executor.userClassPathFirst</code> together.

However, the descriptions of <code>spark.driver.userClassPathFirst</code> misleading us. We thought <code>spark.executor.extraClassPath</code> only works for cluster mode.

The submission command would be something like 

> ... --conf spark.executor.extraClassPath=JarA.jar \
--conf spark.driver.extraClassPath=/path-to-jar/JarA.jar
--conf spark.executor.userClassPathFirst=true
...
--class package.JarAClass
/path-to-jar/JarA.jar

Comment (srowen) [reply docs/configuration.md]: OK. I think it is perhaps implicit that client/cluster mode affects how the driver works, not executors. The real point is below. It's not really causing something to prepend to a classpath, but changes classloader behavior. That sentence doesn't help understand this.

**Comment by HyukjinKwon** [comment]
I think it's unclear what this PR proposes. That's already documented, and both configurations are put together. One is experimental and I think it should be avoided to document an experiment feature to other stable features.

**Comment by lys0716** [comment]
Hi HyukjinKwon, thanks for your reply. What I am trying to do is to improve the document descriptions. Especially how to use customized class in spark cluster.

If you check stackoverflow, there are lots of discussions. But none of them gives a satisfied answer. Even this [one](https://stackoverflow.com/questions/37132559/add-jars-to-a-spark-job-spark-submit) 

We had hard time to test spark on the hadoop cluster. One of the main reason is that we spent a lot of time figuring out how to prepend the customized jar.

I think <code>spark.executor.userClassPathFirst</code> is very helpful in developing, especially you set some customized class path in <code>spark.executor.extraClassPath</code>. It is like <code>setUserClassesTakesPrecedence</code> in hadoop world. 

In the end, if you think we should not mix the experimental and stable feature. I am totally ok with it. I can withdraw this pr.

**Comment by HyukjinKwon** [comment]
I got why those two configurations are important but still one is experimental. Let's document them later together when they become stable. I think we should keep them easily revert-able since they are experimental.

**Comment by srowen** [comment]
These flags are, in practice, not going anywhere and not so experimental. The issue here is isn't that, but that it doesn't explain more about what they do than is already in the doc.

**Pull request closed without merging**